include/zephyr/arch/common/sys_io.h

```diff
@@ -50,6 +50,18 @@ static ALWAYS_INLINE void sys_write32(uint32_t data, mem_addr_t addr)
 	*(volatile uint32_t *)addr = data;
 }
 
+#ifdef CONFIG_64BIT
```

Comment:
Do we really need this? On 32-bit platforms this should be optimized and correctly implemented by the compiler but I don't see why we should not make that available to 32-bit platforms as well.

Reply:
> On 32-bit platforms this should be optimized and correctly implemented by the compiler

we probably dont need this then, updated